train.py

```diff
@@ -11,160 +11,16 @@
-    )
-
-    if config.checkpoint_path is not None:
-        logger.info(f"Loading checkpoint from {config.checkpoint_path}")
-
-        state_dict = cstorch.load(config.checkpoint_path)
-
-        model.load_state_dict(state_dict["model"])
-        if "optimizer" in state_dict:
-            optimizer.load_state_dict(state_dict["optimizer"])
-        if "lr_scheduler" in state_dict:
-            lr_scheduler.load_state_dict(state_dict["lr_scheduler"])
-        global_step = state_dict.get("global_step", 0)
-    else:
-        global_step = 0
-
-    @cstorch.checkpoint_closure
-    def save_checkpoint(step):
-        checkpoint_path = out_dir.joinpath(f"checkpoint_{step}.mdl")
-        state_dict = {
-            "model": model.state_dict(),
-            "optimizer": optimizer.state_dict(),
-            "lr_scheduler": lr_scheduler.state_dict(),
-            "global_step": step,
-            "model_config": asdict(model_config),
-        }
-        cstorch.save(state_dict, checkpoint_path)
-        logger.info(f"Saved checkpoint to {checkpoint_path}")
-
-    @cstorch.trace
-    def training_step(batch):
-        input_ids, labels = batch
-        loss = compiled_model(input_ids, labels)
-        loss.backward()
-        torch.nn.utils.clip_grad_norm_(list(all_params), config.max_gradient_norm)
-        optimizer.step()
-        lr_scheduler.step()
-        optimizer.zero_grad()
-        return loss
-
-    writer = cstorch.utils.tensorboard.SummaryWriter(
-        log_dir=out_dir.joinpath("train")
-    )
-
-    @cstorch.step_closure
-    def log_loss(loss, step):
-        rate = executor.profiler.rate()
-        global_rate = executor.profiler.global_rate()
-
-        logger.info(
-            f"| Step={step}, "
-            f"Loss={loss.item():.5f}, "
-            f"Rate={rate:.2f} samples/sec, "
-            f"GlobalRate={global_rate:.2f} samples/sec"
-        )
-        writer.add_scalar("loss", loss.item(), step)
-        writer.add_scalar("samples_per_second", global_rate, step)
-
-    data_path = os.path.join(config.dataset, "train.bin")
-    dataloader = cstorch.utils.data.DataLoader(
-        get_dataloader,
-        data_path,
-        config.sequence_length,
-        config.batch_size,
-        config.seed,
-    )
-    executor = cstorch.utils.data.DataExecutor(
-        dataloader,
-        num_steps=config.num_steps - global_step,
-        checkpoint_steps=config.checkpoint_steps,
-        cs_config=cs_config,
-        writer=writer,
-    )
-
-    for step, batch in enumerate(executor, start=global_step + 1):
-        if step > config.num_steps:
-            break
-        loss = training_step(batch)
-        log_loss(loss, step)
-        save_checkpoint(step)
-
-    logger.info("Training completed successfully!")
+from cerebras.modelzoo.common.utils.run.cli_pytorch import get_params_from_args
 
+def main():
+    params = get_params_from_args()
+    
+    from cerebras.modelzoo.common.run_utils import main
```

Comment:
do we need to add Cerebras modelzoo dependency in requirements.txt? Please note that this will be a new dependency.